parts/linux/cloud-init/artifacts/cse_helpers.sh

```diff
@@ -485,4 +486,98 @@ process_benchmarks() {
+      release="\"r1804\""
+    elif [[ "${osVersion}" == "20.04" ]] && [[ $(echo "${package}" | jq '.downloadURIs.ubuntu."r2004"') != "null" ]]; then
+      release="\"r2004\""
+    elif [[ "${osVersion}" == "22.04" ]] && [[ $(echo "${package}" | jq '.downloadURIs.ubuntu."r2204"') != "null" ]]; then
+      release="\"r2204\""
+    fi
+  fi
+  echo "${release}"
+}
+
+returnPackageVersions() {
+    local package="$1"
+    local os="$2"
+    local osVersion="$3"
+    local release="$(returnRelease "${package}" "${os}" "${osVersion}")"
+    if [[ "${os}" == "${UBUNTU_OS_NAME}" ]]; then
+        #if .downloadURIs.ubuntu exist, then get the versions from there.
+        #otherwise get the versions from .downloadURIs.default 
+        if [[ $(echo "${package}" | jq ".downloadURIs.ubuntu") != "null" ]]; then
+            versions=$(echo "${package}" | jq ".downloadURIs.ubuntu.${release}.versions[]" -r)
+            for version in ${versions[@]}; do
+             PackageVersions+=("${version}")
+            done
+            return
+        fi
+        versions=$(echo "${package}" | jq ".downloadURIs.default.${release}.versions[]" -r)
+        for version in ${versions[@]}; do
+            PackageVersions+=("${version}")
+        done
+        return
+    fi
+    if [[ "${os}" == "${MARINER_OS_NAME}" ]]; then
+        #if .downloadURIs.ubuntu exist, then get the versions from there.
+        #otherwise get the versions from .downloadURIs.default 
+        if [[ $(echo "${package}" | jq ".downloadURIs.mariner") != "null" ]]; then
+            versions=$(echo "${package}" | jq ".downloadURIs.mariner.${release}.versions[]" -r)
+            for version in ${versions[@]}; do
+                PackageVersions+=("${version}")
+            done
+            return
+        fi
+        versions=$(echo "${package}" | jq ".downloadURIs.default.${release}.versions[]" -r)
+        for version in ${versions[@]}; do
+            PackageVersions+=("${version}")
+        done
+        return    
+    fi
+}
+
+returnPackageDownloadURL() {
+    local package=$1
+    local os=$2
+    local osVersion=$3
+    local release="$(returnRelease "${package}" "${os}" "${osVersion}")"
+    if [[ "${os}" == "${UBUNTU_OS_NAME}" ]]; then
+        #if .downloadURIs.ubuntu exist, then get the downloadURL from there.
+        #otherwise get the downloadURL from .downloadURIs.default 
+        if [[ $(echo "${package}" | jq '.downloadURIs.ubuntu') != "null" ]]; then
+            downloadURL=$(echo "${package}" | jq ".downloadURIs.ubuntu.${release}.downloadURL" -r)
+            echo ${downloadURL}
+            return
+        fi
+        downloadURL=$(echo "${package}" | jq ".downloadURIs.default.${release}.downloadURL" -r)
+        echo ${downloadURL}
+        return    
+    fi
+    if [[ "${os}" == "${MARINER_OS_NAME}" ]]; then
+        #if .downloadURIs.ubuntu exist, then get the downloadURL from there.
```

Comment:
supernit, comment still has downloadURIs.ubuntu

Reply:
Modified the comment to fit the new approach